### PR TITLE
ClangImporter: run Clang with a proper executable path

### PR DESCRIPTION
**Pull request by egorzhdan**
Clang deduces its installation directory from the `argv[0]` parameter (see clang/lib/Frontend/CreateInvocationFromCommandLine.cpp), and the default include search paths are computed based on the installation directory.

This change allows compiling Swift code that imports the C++ stdlib without having to manually specify the include search path of `std` headers.

### CODE REVIEW

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

File: lib/Frontend/Frontend.cpp
Comment (zoecarver) [inline lib/Frontend/Frontend.cpp:517]: Won't this do `build/{ninja|xcode}/swift-{target}/bin/clang` and not `build/{ninja|xcode}/llvm-{target}/bin/clang`? I think we want the latter.

To say this another way, the clang executable will be in the llvm build directory, not the swift build directory.

Comment (egorzhdan) [reply lib/Frontend/Frontend.cpp:517]: Ah you're right, this doesn't help when building the contents of the Swift project itself. I'll think of a better way to do this.

Comment (zoecarver) [reply lib/Frontend/Frontend.cpp:517]: Well, it's not a regression at least, right? So we could potentially land this and then make a follow-up to fix that case.

Comment (egorzhdan) [reply lib/Frontend/Frontend.cpp:517]: Not a regression indeed, but maybe I'll come up with a better solution quickly :) I'll try it this week & will let you know if I don't have another solution.

**Review by zoecarver — APPROVED** [review]

File: lib/Frontend/Frontend.cpp
Comment (zoecarver) [reply lib/Frontend/Frontend.cpp:517]: OK. If you find a better solution I'll happily re-review. If not, I approved so you're welcome to land this.

**Comment by egorzhdan** [comment]
I came up with a better solution that also works when building the Swift project itself.

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@compnerd could you please take a look at the change in CMake build scripts in this PR? I'm wondering if the `-tools-directory` option is passed through `SwiftSource.cmake` correctly.

File: tools/swift-ide-test/swift-ide-test.cpp
Comment (compnerd) [inline tools/swift-ide-test/swift-ide-test.cpp:280]: There is no guarantee that we are using the LLVM tools.  It is acceptable to use binutils (e.g. `gold`, `gas`, etc).

Comment (egorzhdan) [reply tools/swift-ide-test/swift-ide-test.cpp:280]: Ah thanks. The implementation doesn't seem to be affected by that, and I've updated the option description.

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

File: lib/Frontend/CompilerInvocation.cpp
Comment (compnerd) [inline lib/Frontend/CompilerInvocation.cpp:72]: Do we need to worry about the file suffix?  This should be `clang.exe` if running on Windows

Comment (egorzhdan) [reply lib/Frontend/CompilerInvocation.cpp:72]: We're only worried about the directory that the compiler is located in – Clang throws away the last component of the path. It happens here: https://github.com/apple/llvm-project/blob/c5c5f90552428d9012cc0339306eed06ac864e2a/clang/lib/Driver/Driver.cpp#L150]

Before this change, Swift always used `clang` without the `.exe` suffix, and it worked well for Windows.

**Comment by egorzhdan** [comment]
@compnerd @zoecarver folks, could you please take another look at this PR?

File: tools/swift-ide-test/swift-ide-test.cpp
Comment (zoecarver) [inline tools/swift-ide-test/swift-ide-test.cpp:3879]: I guess there's not really a good way to test this path...

**Review by zoecarver — APPROVED** [review]
Still looks good to me! Thanks.

**Comment by zoecarver** [comment]
Might be good to make sure @compnerd is OK with this. It seems more in his domain. 

Please at least make sure all the tests still pass (including windows). 

@swift-ci please test.

**Comment by zoecarver** [comment]
@swift-ci please test.

File: tools/swift-ide-test/swift-ide-test.cpp
Comment (egorzhdan) [reply tools/swift-ide-test/swift-ide-test.cpp:3879]: Yeah, and since this is only used by Swift tests I think it should be fine to leave as is.

**Comment by egorzhdan** [comment]
The tests are passing, and I'm going to merge this to unblock other changes related to `std` importing.
I'm happy to address any feedback after merging.

**Pull request merged**
